framework/src/auxkernels/ArrayParsedAux.C

```diff
@@ -84,9 +84,10 @@ ArrayParsedAux::ArrayParsedAux(const InputParameters & parameters)
   setParserFeatureFlags(_func_F);
 
   // add the constant expressions
-  addFParserConstants(_func_F,
-                      getParam<std::vector<std::string>>("constant_names"),
-                      getParam<std::vector<std::string>>("constant_expressions"));
+  if (isParamValid("constant_names") && isParamValid("constant_expressions"))
```

Comment:
try empty default instead

Reply:
Do we want to go through all the `isParamValid` conditions to see if it can be replaced by default value? I've tested most of them with default value and then went back to add conditions while working on this PR. There may be a few places that it can be replaced by default value but definitely not a lot.  Feel like we are repeating all that work now and it will be very time consuming to test each changes. The current change doesn't break any test cases or apps. Could we accept it as it is?

Reply:
It s not really a matter of seeing if you can, it s a matter that it should be possible. And if it's not we need to know why. It was certainly possible before since there was a default value. Something else must have changed

Reply:
I think it is beyond the scope of this PR. There are definitely places to improve all the downstream logics but for this PR, I want to make sure all the existing cases works with this new patch. And for all the new codes there won't be any addParam with implicit default values. Otherwise I'll have to keep patching them

Reply:
is there not value in different logic for when a user provides a blank vs a user providing nothing and having a default blank?

Reply:
yeah if we care about users doing `some_vector_param = ""` then any previous (before this PR) isParamValid logic will have to rely on the size of the vector instead.
There still wont be a need for new isParamValid logic